watertap/unit_models/steam_heater_0D.py

```diff
@@ -0,0 +1,151 @@
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+from pyomo.environ import Var
+
+from idaes.core import (
+    declare_process_block_class,
+)
+from idaes.models.unit_models.heat_exchanger import HeatExchangerData
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+import idaes.logger as idaeslog
+from watertap.costing.unit_models.heat_exchanger import (
+    cost_heat_exchanger,
+)
+from pyomo.common.config import ConfigValue
+
+
+_log = idaeslog.getLogger(__name__)
+
+
+__author__ = "Elmira Shamlou"
+
+"""
+This unit model uses is based on the IDAES `feedwater_heater_0D` model.
+However, the constraints and properties defined in the IDAES unit model do not align with those in the available WaterTAP property packages.
+To address this, alternative constraints have been replaced to ensure full condensation based on the WaterTAP properties.
+Note that additional components like desuperheaters, drain mixers, and coolers are not included. If necessary, these can be modeled separately by adding heat exchangers and mixers to the flowsheet.
+To do: Consider incorporating as a modification to IDAES' FeedWaterHeater model directly in the IDAES repo"
+"""
+
+
+@declare_process_block_class(
+    "SteamHeater0D",
+    doc="""Feedwater Heater Condensing Section
+The feedwater heater condensing section model is a normal 0D heat exchanger
+model with an added constraint to calculate the steam flow such that the outlet
+of shell is a saturated liquid.""",
+)
+class SteamHeater0DData(HeatExchangerData):
+    CONFIG = HeatExchangerData.CONFIG()
+    CONFIG.declare(
+        "has_saturation_pressure_deviation",
+        ConfigValue(
+            default=False,
+            domain=bool,
+            description="Flag to indicate if saturation pressure deviation should be considered",
+            doc="""Indicates whether the saturation pressure deviation at the outlet of the steam heater should be
+               modeled. If True, 'saturation_pressure_deviation' needs to be specified by the user.""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.hot_side.config.property_package.component_list,
+            doc="Mass balance",
+        )
+        def outlet_liquid_mass_balance(b, t, j):
+            b.hot_side.properties_out[t].flow_mass_phase_comp["Vap", j].fix(0)
+            return (
+                b.hot_side.properties_in[t].flow_mass_phase_comp["Vap", j]
+                + b.hot_side.properties_in[t].flow_mass_phase_comp["Liq", j]
+                == b.hot_side.properties_out[t].flow_mass_phase_comp["Liq", j]
+            )
```

Comment:
I don't quite understand why you need to have an extra mass balance. It seems to duplicate the pressure saturation constraints later. Additionally, it's not very good practice to fix any variable as 0, especially at the unit model level.

Reply:
@Morgan88888888 WaterTAP property packages require modeling the saturation condition differently than IDAES, where IAPWS is used for steam properties. In the case of the WaterTAP property package, it's necessary to ensure a liquid state at the outlet through mass balance. However, maintaining this mass balance alone doesn't guarantee the saturation condition. Therefore, we also need to impose the saturation condition using saturation pressure. Similarly, enforcing the saturation pressure condition alone does not guarantee that the outlet remains in a liquid state, necessitating the inclusion of both saturation pressure and mass balance to meet these conditions.